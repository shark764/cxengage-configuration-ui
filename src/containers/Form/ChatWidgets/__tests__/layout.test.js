/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';

import ChatWidgetsForm from '../layout';

const initialValues = {
  get: () => {}
};

describe('<ChatWidgetsForm />', () => {
  it('renders create chatWidgets', () => {
    const rendered = shallow(
      <ChatWidgetsForm
        name="mockName"
        shared={false}
        welcome="hero"
        fontFamily="arial"
        fontSize="12pt"
        headerColor="#000000"
        headerTextIcons="#000000"
        chatbg="#000000"
        agentHeader="#000000"
        agentText="#000000"
        customerHeader="#000000"
        customerText="#000000"
        systemText="#000000"
        buttonText="#000000"
        iconColor="#000000"
        borderColor="#000000"
        handleSubmit={() => {}}
        initialValues={initialValues}
        digitalChannelsAppsFetching={false}
        digitalChannelsAppIds={[{ value: 'mockValue', label: 'mockLabel' }]}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
  it('renders update chatWidgets', () => {
    const rendered = shallow(
      <ChatWidgetsForm
        name="mockName"
        shared={false}
        welcome="hero"
        fontFamily="arial"
        fontSize="12pt"
        headerColor="#000000"
        headerTextIcons="#000000"
        chatbg="#000000"
        agentHeader="#000000"
        agentText="#000000"
        customerHeader="#000000"
        customerText="#000000"
        systemText="#000000"
        buttonText="#000000"
        iconColor="#000000"
        borderColor="#000000"
        whitelistedUrls={['http://www.example.com']}
        handleSubmit={() => {}}
        initialValues={initialValues}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
});
