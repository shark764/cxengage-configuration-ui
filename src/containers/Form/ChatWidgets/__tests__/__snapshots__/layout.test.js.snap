// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatWidgetsForm /> renders create chatWidgets 1`] = `
<form
  onSubmit={[Function]}
>
  <InputField
    className="name-input-field"
    componentType="input"
    disabled={false}
    inputType="text"
    label="Name *"
    name="name"
  />
  <ToggleField
    className="shared-toggle-field"
    disabled={false}
    label="Shared"
    name="shared"
  />
  <DetailHeader
    text="Fields"
  />
  <ListField
    className="list-inputs-field-chat-widgets"
    disabled={false}
    label="Inputs"
    name="inputs"
  />
  <DetailHeader
    text="Images"
  />
  <RadioGroupField
    className="radio-group-field-chat-widgets"
    disabled={false}
    label="Welcome"
    name="welcome"
    options={
      Array [
        Object {
          "helpText": "(Display 300 x 410)",
          "label": "Hero",
          "value": "hero",
        },
        Object {
          "helpText": "(Display 300 x 800)",
          "label": "Logo",
          "value": "logo",
        },
      ]
    }
    required={true}
  />
  <FileUpload
    acceptedFileType=".jpg"
    className="file-upload-field"
    id="dtpanel-chat-widget-upload-csv"
  />
  <DetailHeader
    text="Type"
  />
  <SelectField
    className="font-input-select-field"
    disabled={false}
    label="Font Family"
    name="fontFamily"
    options={
      Array [
        Object {
          "label": "Arial",
          "value": "arial",
        },
      ]
    }
    required={false}
  />
  <SelectField
    className="font-input-select-field"
    disabled={false}
    label="Font Size"
    name="fontSize"
    options={
      Array [
        Object {
          "label": "12pt",
          "value": "12pt",
        },
      ]
    }
    required={false}
  />
  <DetailHeader
    text="Color"
  />
  <ColorField
    className="chat-widgets-header-color"
    disabled={false}
    label="Header Color"
    name="headerColor"
  />
  <ColorField
    className="chat-widgets-header-text-icons"
    disabled={false}
    label="Header Text/Icons"
    name="headerTextIcons"
  />
  <ColorField
    className="chat-widgets-chat-background"
    disabled={false}
    label="Chat BG"
    name="chatbg"
  />
  <ColorField
    className="chat-widgets-agent-header"
    disabled={false}
    label="Agent Header"
    name="agentHeader"
  />
  <ColorField
    className="chat-widgets-agent-text"
    disabled={false}
    label="Agent Text"
    name="agentText"
  />
  <ColorField
    className="chat-widgets-customer-header"
    disabled={false}
    label="Customer Header"
    name="customerHeader"
  />
  <ColorField
    className="chat-widgets-customer-text"
    disabled={false}
    label="Customer Text"
    name="customerText"
  />
  <ColorField
    className="chat-widgets-system-text"
    disabled={false}
    label="System Text"
    name="systemText"
  />
  <ColorField
    className="chat-widgets-button-text"
    disabled={false}
    label="Button Text"
    name="buttonText"
  />
  <ColorField
    className="chat-widgets-icon-color"
    disabled={false}
    label="Icon Color"
    name="iconColor"
  />
  <ColorField
    className="chat-widgets-border-color"
    disabled={false}
    label="Border Color"
    name="borderColor"
  />
  <DetailHeader
    text="Whitelist"
  />
  <ListField
    className="list-urls-field-chat-widgets"
    disabled={false}
    label="Urls"
    name="urls"
  />
</form>
`;
